[nanodbc] try to link with unixodbc from system

****

**Issue**
On a WSL ubuntu 18.04 project using `nanodbc` don't link.

**Reproductibility**
- Have vcpkg installed with `nanodbc`.
- Have a sample project linking with `nanodbc`.

**Investigations**
- When installing component from vcpkg, it compile the dependency `unixodbc`.
- When a project link with `nanodbc` the project add a link `-lodbc` [nanodbc cmake](https://github.com/nanodbc/nanodbc/blob/fe1d590991da30dc9cb71676c4d80cb2d9acb49e/CMakeLists.txt#L118) 
- From Visual studio we receive a link error `/usr/bin/ld: cannot find -lodbc`

**Workaround**
It's not a workaround I'm satisfied with, but it unlock my build on ubuntu
```
sudo apt-get install unixodbc-dev
```

**Expected result**
Link only with `unixodbc` compiled from vcpkg and nothing from the system.
Guess there should be some extra steps in portfile but don't know how to do them.

--- a/ports/nanodbc/find-unixodbc.patch
+++ b/ports/nanodbc/find-unixodbc.patch
@@ -1,0 +1,24 @@
+--- "a/CMakeLists.txt"
++++ "b/CMakeLists.txt"
+@@ -110,6 +110,13 @@ message(STATUS "nanodbc feature: Enable SQL_NO_DATA bug workaround - ${NANODBC_E
+ ## find unixODBC or iODBC config binary
+ ########################################
+ if(UNIX)
++  # Try to find unixodbc package first
++  find_package(unixodbc)
++  if(unixodbc_FOUND)
++    message(STATUS "nanodbc build: unixODBC package found")
++    set(ODBCLIB UNIX::odbc)
++    set(ODBC_CONFIG true)
++  else()
+   # Try to find unixODBC first via odbc_config program.
+   find_program(ODBC_CONFIG odbc_config
+     PATHS $ENV{ODBC_PATH}/bin /usr/bin /usr/local/bin)
+@@ -158,6 +165,7 @@ if(UNIX)
+       endif()
+     endif()
+   endif()
++  endif()
+ 
+   if(NOT ODBC_CONFIG)
+     message(FATAL_ERROR "can not find a suitable odbc driver manager")

--- a/ports/nanodbc/portfile.cmake
+++ b/ports/nanodbc/portfile.cmake
@@ -11,6 +11,7 @@ vcpkg_from_github(
     PATCHES
         rename-version.patch
         add-missing-include.patch
+        find-unixodbc.patch
 )
 file(RENAME "${SOURCE_PATH}/VERSION" "${SOURCE_PATH}/VERSION.txt")
 

--- a/ports/nanodbc/vcpkg.json
+++ b/ports/nanodbc/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "nanodbc",
   "version": "2.13.0",
-  "port-version": 5,
+  "port-version": 6,
   "description": "A small C++ wrapper for the native C ODBC API.",
   "homepage": "https://github.com/nanodbc/nanodbc",
   "dependencies": [

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4570,7 +4570,7 @@
     },
     "nanodbc": {
       "baseline": "2.13.0",
-      "port-version": 5
+      "port-version": 6
     },
     "nanoflann": {
       "baseline": "1.3.2",

--- a/versions/n-/nanodbc.json
+++ b/versions/n-/nanodbc.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "7de6296fc337f181517f8ed4c2e6ed35d749e414",
+      "version": "2.13.0",
+      "port-version": 6
+    },
+    {
       "git-tree": "82bfc56de1430aa6fec9c27925d46a72e1b800a0",
       "version": "2.13.0",
       "port-version": 5

